From 72b23883f2984822ce11f7fa90b43541f4f69dff Mon Sep 17 00:00:00 2001
From: Martin Hrdlicka <martin.hrdlicka@nxp.com>
Date: Thu, 2 Mar 2023 17:10:26 +0100
Subject: [PATCH 63/63] dt-bindings: net: pfe: update dt bindings documentation

Remove global property nxp,pfeng-hif-channels
The property is not needed as the same can be calculated
from netif property.

Netdev property nxp,pfeng-emac-id renamed to nxp,pfeng-linked-phyif
Preparation for netdev linked with HIF.

Issue: AAVB-6542

Upstream-Status: Pending 

Signed-off-by: Jan Petrous <jan.petrous@nxp.com>
Signed-off-by: Martin Hrdlicka <martin.hrdlicka@nxp.com>
Signed-off-by: Zhantao Tang <zhantao.tang@windriver.com>
---
 .../devicetree/bindings/net/nxp,s32g-pfe.yaml | 67 ++++++++++++-------
 1 file changed, 44 insertions(+), 23 deletions(-)

diff --git a/Documentation/devicetree/bindings/net/nxp,s32g-pfe.yaml b/Documentation/devicetree/bindings/net/nxp,s32g-pfe.yaml
index 08229d015b97..a4e41e032cb4 100644
--- a/Documentation/devicetree/bindings/net/nxp,s32g-pfe.yaml
+++ b/Documentation/devicetree/bindings/net/nxp,s32g-pfe.yaml
@@ -1,5 +1,5 @@
 # SPDX-License-Identifier: (GPL-2.0-only OR BSD-2-Clause)
-# Copyright 2022 NXP
+# Copyright 2022-2023 NXP
 %YAML 1.2
 ---
 $id: "http://devicetree.org/schemas/net/nxp,s32g-pfe.yaml#"
@@ -117,28 +117,29 @@ properties:
       firmware search path containing the firmware image used when
       initializing PFE UTIL hardware. Optional.
 
-  nxp,pfeng-hif-channels:
+  nxp,pfeng-emac-ts-ext-modes:
     $ref: /schemas/types.yaml#/definitions/uint32-array
     description: |
-      The set of managed HIF channels. Any combination of standard
-      HIF channel can be used. HIF_NCPY is not supported.
-    minItems: 1
-    maxItems: 4
-    items:
-      minimum: 0
-      maximum: 3
+      The set of PFE_EMACs required to work in external timestamping
+      mode. The combination of external TS support is limited,
+      check S32G Reference Manual for detailed info.
+    maxItems: 3
+    contains:
+      enum:
+        - PFE_PHYIF_EMAC_0
+        - PFE_PHYIF_EMAC_1
+        - PFE_PHYIF_EMAC_2
 
   nxp,pfeng-ihc-channel:
     $ref: /schemas/types.yaml#/definitions/uint32
     description: |
       The HIF channel number used for IHC transport.
-      The outgoing channel, must be one from set
-      specified in nxp,pfeng-hif-channels.
     maxItems: 1
-    items:
-      minimum: 0
-      maximum: 3
-
+      enum:
+        - PFE_PHYIF_HIF_0
+        - PFE_PHYIF_HIF_1
+        - PFE_PHYIF_HIF_2
+        - PFE_PHYIF_HIF_3
 
   nxp,pfeng-master-channel:
     $ref: /schemas/types.yaml#/definitions/uint32
@@ -148,9 +149,12 @@ properties:
       receiving IHC messages. Here can also be used
       the HIF_NCPY channel.
     maxItems: 1
-    items:
-      minimum: 0
-      maximum: 4
+      enum:
+        - PFE_PHYIF_HIF_0
+        - PFE_PHYIF_HIF_1
+        - PFE_PHYIF_HIF_2
+        - PFE_PHYIF_HIF_3
+        - PFE_PHYIF_HIF_NOCPY
 
   phys:
     maxItems: 3
@@ -203,13 +207,30 @@ properties:
           HIF channel can be used. HIF_NOCPY is not supported.
         minItems: 1
         maxItems: 4
-        items:
-          minimum: 0
-          maximum: 3
+          enum:
+            - PFE_PHYIF_HIF_0
+            - PFE_PHYIF_HIF_1
+            - PFE_PHYIF_HIF_2
+            - PFE_PHYIF_HIF_3
       nxp,pfeng-emac-id:
-        $ref: /schemas/types.yaml#/definitions/uint32-array
+        $ref: /schemas/types.yaml#/definitions/uint32
+        description: |
+          The linked PFE_EMAC port. Use nxp,pfeng-linked-phyif instead.
+        deprecated: true
+      nxp,pfeng-linked-phyif:
+        $ref: /schemas/types.yaml#/definitions/uint32
         description: |
-          The linked PFE_EMAC port.
+          The linked PFE physical port.
+        maxItems: 1
+          enum:
+            - PFE_PHYIF_EMAC_0
+            - PFE_PHYIF_EMAC_1
+            - PFE_PHYIF_EMAC_2
+            - PFE_PHYIF_HIF_NOCPY
+            - PFE_PHYIF_HIF_0
+            - PFE_PHYIF_HIF_1
+            - PFE_PHYIF_HIF_2
+            - PFE_PHYIF_HIF_3
       nxp,pfeng-netif-mode-aux:
         $ref: /schemas/types.yaml#/definitions/flag
         description: |
-- 
2.25.1

